### PR TITLE
V21

### PR DESCRIPTION
**Pull request by 0xmurloc**
This branch will be our v2.1 feature branch. 
- copied v2 docs to separate `v2` section 
- updated concepts for v2.1

TODO
- [ ] Review `Guides` and `SDK` examples
- [ ] Update deployment addresses
- [ ] Update subgraphs
- [ ] Add link to v2.1 audit report
- [ ] Add DexBarn API docs

### CODE REVIEW

File: docs/guides/decode-examples.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+sidebar_label: Decode Examples
+---
+
+## Decode Bytes Examples
+
+Some of the new output variables from `functions` and `event` of the new `v2.1 Dex` needs to be decoded. Here are some propositions in order to do so: <br/>
+
+### Decode `amountsOut` from the `Swap` event
+
+The following functions return the decoded values (x, y) extracted from the encoded `bytes32` amounts. The functions are returning the results in a `tuple`.
```
Comment (cryptofish7) [inline docs/guides/decode-examples.md:12]: The parameter `amountsIn` is packed with the amounts in for both tokens X and Y and encoded into `bytes32`. Below is an example of how to decode them. The same is true for `amountsOut`.

File: docs/guides/decode-examples.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+```typescript
+function decodeAmounts(amounts: Bytes): [bigint, bigint] {
+  /**
+   * Decodes the amounts bytes input as 2 integers.
+   *
+   * @param amounts - amounts to decode.
+   * @return tuple of BigInts with the values decoded.
+   */
```
Comment (cryptofish7) [inline docs/guides/decode-examples.md:21]: New line after this line.

File: docs/guides/decode-examples.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  // Convert amounts to a BigInt
+  const amountsBigInt = BigInt(`0x${Buffer.from(amounts).toString('hex')}`);
+
+  // Read the first 128 bits of the 256 bits
```
Comment (cryptofish7) [inline docs/guides/decode-examples.md:25]: // Read the right 128 bits of the 256 bits

File: docs/guides/decode-examples.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    :param amounts: amounts to decode.
+    :return: tuple of ints with the values decoded.
+    """
+    amounts = bytes.fromhex(amounts)
+    # read the first 128bits of the 256bits
+    amountsX = int.from_bytes(random_bytes, byteorder="big") & (2 ** 128 - 1)
+    # read the last 128bits of the 256bits
+    amountsY = int.from_bytes(random_bytes, byteorder="big") >> 128
+    return (amountsX, amountsY)
+```
```
Comment (cryptofish7) [inline docs/guides/decode-examples.md:49]: Remove.

File: docs/guides/decode-examples.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    """
+    Decode the Fee value from the bytes input and return it into an int.
+    Fee values are stored in the first 128bits ofthe input.
+
+    :param fees_bytes: Bytes containing the encoded value.
+    :return: Fee values.
+    """
+    fees_bytes = bytes.fromhex(fees_bytes)
+    return int.from_bytes(fees_bytes, byteorder="big") & (2 ** 128 - 1)
+```
```
Comment (cryptofish7) [inline docs/guides/decode-examples.md:83]: Remove.

File: docs/guides/decode-examples.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    amountsY = int.from_bytes(random_bytes, byteorder="big") >> 128
+    return (amountsX, amountsY)
+```
+
+### Decode `totalFees` and `protocolFees` from `Swap` Event
+
+The following functions return the decoded values of the Fees of the `LBPair` contract.
+
+```typescript
+function decodeFees(feesBytes: Bytes): bigint {
```
Comment (cryptofish7) [inline docs/guides/decode-examples.md:56]: Still confused from this function. What refers to `totalFees` and what refers to `protocolFees`?

File: docs/guides/decode-examples.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  /**
+   * Decode the Fee value from the bytes input and return it into a BigInt.
```
Comment (cryptofish7) [inline docs/guides/decode-examples.md:58]: Decode the fee value from the bytes input and return it as in integer.

File: docs/guides/decode-examples.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+   * Fee values are stored in the first 128 bits of the input.
+   *
+   * @param feesBytes - Bytes containing the encoded value.
+   * @return Fee values.
+   */
```
Comment (cryptofish7) [inline docs/guides/decode-examples.md:63]: Newline after this line.

File: docs/guides/decode-examples.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+  // Convert feesBytes to a BigInt
+  const feesBigInt = BigInt(`0x${Buffer.from(feesBytes).toString('hex')}`);
+
+  // Retrieve the fee value from the first 128 bits
```
Comment (cryptofish7) [inline docs/guides/decode-examples.md:67]: // Retrieve the fee value from the right 128 bits

**Review by cryptofish7 — CHANGES_REQUESTED** [review]
@0xmurloc rearranged the folder structure but it's basically replicated all the files under V2.0. 

Please remove that entire V2.0 folder.

File: docs/guides/decode-examples.md (hunk 5 above)
Comment (0xmurloc) [reply docs/guides/decode-examples.md:83]: python example is very useful for integration partners, they dont index with solidity.

**Comment by 0xmurloc** [comment]
> @0xmurloc rearranged the folder structure but it's basically replicated all the files under V2.0.
> 
> Please remove that entire V2.0 folder.

The contracts and libraries docs are updated in v2.1. We maintain v2.0 folder for users that may still need to refer to v2.0 contracts. It may take some time for all v2 usage to fully migrate to v2.1 (E.g. even today we still have alot of v1 pools). So we will need to maintain and support both v2 and v2.1.

**Pull request merged**